### PR TITLE
feat: function to estimate thor trade fees

### PR DESCRIPTION
**Pull request by elmutt**
Util function to estimate thor trade fees for any eth/erc20/btc fee.

We have no way of knowing the exact TX thorchain is going to make and how much gas/utxo bytes it will use but we can hardcode very close estimates and this works good enough (theres no other way to do it as far as I know)

### CODE REVIEW

**Review by DaoDev44 — APPROVED** [review]

**Review by DaoDev44 — DISMISSED** [review]

File: packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-console */
```
Comment (0xdef1cafe) [inline packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.ts:1]:
```suggestion
```

File: packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  if (!inboundInfo)
+    throw new SwapError('[estimateTradeFee] - unable to locate inbound pool info', {
+      code: SwapErrorTypes.VALIDATION_FAILED,
+      details: { thorPoolChainId }
+    })
+
+  const gasRate = inboundInfo.gas_rate
+  const { chainId, assetNamespace } = fromAssetId(buyAssetId)
+
+  const tradeFee = (() => {
```
Comment (0xdef1cafe) [inline packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.ts:53]: why the IIFE here if we're not awaiting?

File: packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.test.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import { mockInboundAdresses } from '../test-data/midgardResponse'
+import { thorService } from '../thorService'
+import { estimateTradeFee } from './estimateTradeFee'
+
+jest.mock('../thorService')
+
+describe('estimateTradeFee', () => {
+  const deps = { midgardUrl: 'localhost:3000', adapterManager: <ChainAdapterManager>{} }
+  it('should correcty estimate a trade fee for bitcoin', async () => {
```
Comment (0xdef1cafe) [inline packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.test.ts:11]:
```suggestion
  it('should correctly estimate a trade fee for bitcoin', async () => {
```

File: packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.test.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    )
+    const estimatedTradeFee = await estimateTradeFee(
+      deps,
+      'bip122:000000000019d6689c085ae165831e93/slip44:0'
+    )
+
+    const expectedResult = '16362'
+    expect(estimatedTradeFee).toEqual(expectedResult)
+  })
+  it('should correcty estimate a trade fee for ethereum', async () => {
```
Comment (0xdef1cafe) [inline packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.test.ts:23]:
```suggestion
  it('should correctly estimate a trade fee for ethereum', async () => {
```

File: packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.test.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    ;(thorService.get as jest.Mock<unknown>).mockReturnValue(
+      Promise.resolve({ data: mockInboundAdresses })
+    )
+    const estimatedTradeFee = await estimateTradeFee(deps, 'eip155:1/slip44:60')
+
+    const expectedResult = '32241720000000000'
+    expect(estimatedTradeFee).toEqual(expectedResult)
+  })
+  it('should correcty estimate a trade fee for an ethereum erc20 asset', async () => {
```
Comment (0xdef1cafe) [inline packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.test.ts:32]:
```suggestion
  it('should correctly estimate a trade fee for an ethereum erc20 asset', async () => {
```

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.ts (hunk 2 above)
Comment (elmutt) [reply packages/swapper/src/swappers/thorchain/utils/estimateTradeFee/estimateTradeFee.ts:52]: talked with defi ok discord. will change

**Review by 0xdef1cafe — APPROVED** [review]

**Review by DaoDev44 — APPROVED** [review]

**Pull request merged**